src/components/structures/MatrixChat.js

```diff
@@ -891,6 +895,7 @@ module.exports = React.createClass({
         });
 
         cli.on('sync', function(state, prevState) {
+            dis.dispatch({action: 'sync_state', prevState, state});
```

Comment:
Eugh, sounds like the general impedance mismatch of the two different dispatch mechanisms, but also sounds like this is the correct way to get around it. A comment might not be a bad idea so you don't have to have this conversation with the next 8 people who read that line.